smartsim/_core/mli/comm/channel/dragonchannel.py

```diff
@@ -69,10 +104,16 @@ def from_descriptor(
     ) -> "DragonCommChannel":
         """A factory method that creates an instance from a descriptor string
 
-        :param descriptor: The descriptor that uniquely identifies the resource
+        :param descriptor: The descriptor that uniquely identifies the resource. Output
+        from `descriptor_string` is correctly encoded.
         :returns: An attached DragonCommChannel"""
         try:
-            return DragonCommChannel(base64.b64decode(descriptor))
+            utf8_descriptor = descriptor.encode("utf-8")
+            actual_descriptor = base64.b64decode(utf8_descriptor)
+            channel = dch.Channel.attach(actual_descriptor)
+            return DragonCommChannel(channel)
```

Comment:
I get it now!

Reply:
I started at the bottom and wondered "what does she get!?" - now i get what you get...